pipeline/detection.py

```diff
@@ -110,10 +113,40 @@ def run(self, *args, **kwargs):
         return ds
 
     def extract_sources(self, image):
+        """
+        Run source-extraction (using SExtractor) on the given image.
+
+        Parameters
+        ----------
+        image: Image
+            The image to extract sources from.
+
+        Returns
+        -------
+        sources: SourceList
+            The list of sources detected in the image.
+            This contains a table where each row represents
+            one source that was detected, along with all its properties.
+
+        """
         # TODO: finish this
-        # TODO: this should also generate an estimate of the PSF
+        # TODO: this should also generate an estimate of the PSF?
+
+        data = image.data
+
+        # see the note in https://sep.readthedocs.io/en/v1.0.x/tutorial.html#Finally-a-brief-word-on-byte-order
+        if data.dtype == '>f8':  # TODO: what about other datatypes besides f8?
+            data = data.byteswap().newbyteorder()
+        b = sep.Background(data)
```

Comment:
Thought is going to be required here.  My experience is that you want to use different sky-subtraction algorithms in different places.  For extragalactic fields, the default sextractor algoirithm usually does a great job.   (Is that exact same algorithm what sep.Background uses by default?)  However, for crowded galactic fields, it oversubtracts the sky by a lot, and a slower and more complicated algorithm does better; the one I've used is in Bijaoui, 1980, A&A, 84, 81

Reply:
Another thought : we don't always want to subtract a background.  The most obvious case is going to be where the image is a subtraction!  But we might also end up with sky-subtracted images already (e.g. if there's a custom processing pipeline that really does a good job of subtracting sky images, and we don't want to mess it up by doing something on top of that here).  Perhaps a field in the image table that indicates if the image is already background subtracted?

Reply:
I like having that extra field. Maybe even worthwhile to save the median sky brightness and/or the sky noise level so we will not have to re-estimate it. 
The thing is that even if you reduce the mean sky (or the local sky value) it still leaves the noise. If you get an Image that is already subtracted but you don't know how much was there before, you'd be missing a big piece of information which is the sky variance. 
This seems like we should make an issue with a "decision needed" on it and discuss on the next meeting.

Reply:
yeah, I agree.